File: spark/src/main/java/com.spark/java/sparkCore/FileRDD.java
package com.spark.java.sparkCore;

import org.apache.spark.SparkConf;
import org.apache.spark.api.java.JavaRDD;
import org.apache.spark.api.java.JavaSparkContext;

public class FileRDD {
    public static void main(String[] args) {
        SparkConf conf = new SparkConf();
        conf.setAppName("SparkArrayTest")
                .setMaster("local");
        JavaSparkContext sc = new JavaSparkContext(conf);

        JavaRDD<String> file = sc.textFile("data\\words.txt");

        JavaRDD<Integer> map = file.map(e -> e.length());

        Integer res = map.reduce((x, y) -> x + y);

        System.out.println(res);
    }
}

File: spark/src/main/java/com.spark/scala/FileRDD.scala
package com.spark.scala

import org.apache.spark.{SparkConf, SparkContext}

object FileRDD {
  def main(args: Array[String]): Unit = {
    val conf = new SparkConf()
    conf.setAppName("FileRDD")
      .setMaster("local")
    val sc = new SparkContext(conf);

    val fileRDD = sc.textFile("data\\words.txt", 2)

    val res = fileRDD.flatMap(_.split(" ")).map((_, 1)).reduceByKey((x, y) => x + y).sortBy(_._2)

    res.foreach( e => println(e._1 + "~" + e._2))
  }
}
